### PR TITLE
Implement SA1201

### LINKED ISSUE #61: SA1201: ElementsMustAppearInTheCorrectOrder
http://www.stylecop.com/docs/SA1201.html

### PR DESCRIPTION
**Pull request by pdelvo**
Fixes #61. i have not yet implemented a code fix because I think we should discuss how it should fix a violation.

### CODE REVIEW

**Comment by sharwell** [comment]
I created issue #573 for the code fix.

File: StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1201ElementsMustAppearInTheCorrectOrder.cs [hunk 1]
```diff
@@ -103,17 +107,40 @@ public class SA1201ElementsMustAppearInTheCorrectOrder : DiagnosticAnalyzer
...
 
         private static readonly DiagnosticDescriptor Descriptor =
-            new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, AnalyzerConstants.DisabledNoTests, Description, HelpLink);
+            new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, Category, DiagnosticSeverity.Warning, true, Description, HelpLink);
 
         private static readonly ImmutableArray<DiagnosticDescriptor> SupportedDiagnosticsValue =
             ImmutableArray.Create(Descriptor);
 
+        // extern alias and usings are missing here because the compiler itself is enforcing the right order.
+        private readonly static ImmutableArray<SyntaxKind> OuterOrder = ImmutableArray.Create(
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1201ElementsMustAppearInTheCorrectOrder.cs]: :bulb: [SA1206](http://www.stylecop.com/docs/SA1206.html) says `private static readonly`.

**Comment by sharwell** [comment]
I don't like how a nested class at the beginning of a class can result in *every other element* getting a warning. What would happen if we are more selective in our reporting?

:memo: Only one of the following items would be used.

1. Report a warning on the current element if it should come after the element that follows it.
2. Report a warning on the current element if it should appear before the element that precedes it.

**Comment by pdelvo** [comment]
I implemented the change you suggested. The diagnostics that get reported are now: A X should not follow a Y.

**Comment by pdelvo** [comment]
I need to work on thisd a bit more to remove the compiler errors in the unit tests. I also just discovered a crash if the document is in an invalid state and has an IncompleteMemberSyntax in it

**Comment by sharwell** [comment]
:question: Is this ready now?

**Comment by pdelvo** [comment]
I think so.

**Pull request merged**